haystack/document_stores/milvus2x.py

```diff
@@ -551,22 +490,17 @@ def delete_documents(
         if headers:
             raise NotImplementedError("Milvus2DocumentStore does not support headers.")
 
-        index = index or self.index
-        super().delete_documents(index=index, filters=filters)
-
-        connection = connections.get_connection()
-        has_collection = connection.has_collection(collection_name=index)
-        if not has_collection:
-            logger.warning("No index exists. Use 'update_embeddings()` to create an index.")
+        if ids:
+            self._delete_vector_ids_from_milvus(ids=ids, index=index)
+        elif filters:
+            existing_docs = super().get_all_documents(filters=filters, index=index)
```

Comment:
Is it possible to use Generator?
If is it loose filter then it going to consume a lot of memory as well processing.

Reply:
That sounds like a good idea. The feature is now added.